### PR TITLE
New community landing page

### PR DESCRIPTION
**Pull request by wushenshen**
#704 

Needs https://github.com/Hylozoic/hylo-node/pull/586

### CODE REVIEW

File: src/components/FeedList/FeedList.js [hunk 1]
```diff
@@ -95,8 +95,8 @@ export default class FeedList extends React.Component {
     const stickyTabBarStyle = {
       width: tabBarWidth + 'px'
     }
-    const isProject = routeParams.postTypeContext === 'project'
-    const isEvent = routeParams.postTypeContext === 'event'
+    const isProject = routeParams.postTypeContext || routeParams.view === 'project'
```
Comment (tibetsprague) [inline src/components/FeedList/FeedList.js:98]: this doesnt seem right. is it supposed to be &&?

File: src/components/PostEditor/PostEditor.connector.js [hunk 2]
```diff
@@ -62,7 +61,7 @@ export function mapStateToProps (state, props) {
   const networkSlug = getRouteParam('networkSlug', null, props)
   const topic = getTopicForCurrentRoute(state, props)
   const topicName = get('name', topic)
-  const postTypeContext = getPostTypeContext(null, props) || getQuerystringParam('t', null, props)
```
Comment (tibetsprague) [inline src/components/PostEditor/PostEditor.connector.js:65]: hmm what was the t query param used for..??

File: src/routes/CommunitySidebar/CommunitySidebar.js [hunk 3]
```diff
@@ -24,6 +24,8 @@ export default class CommunitySidebar extends Component {
   render () {
     const { community, members, leaders, canModerate } = this.props
 
+    if (community.memberCount === 0) return null
```
Comment (tibetsprague) [inline src/routes/CommunitySidebar/CommunitySidebar.js:27]: why is this needed? it should certainly be below the next conditional which checks if community exists.

File: src/routes/LandingPage/LandingPage.connector.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import getRouteParam from 'store/selectors/getRouteParam'
+import presentCommunity from 'store/presenters/presentCommunity'
+
+export function mapStateToProps (state, props) {
+  let community, communityTopic
+  const communitySlug = getRouteParam('slug', state, props)
+
+  if (communitySlug) {
+    community = presentCommunity(getCommunityForCurrentRoute(state, props))
+    // communityTopic = getCommunityTopicForCurrentRoute(state, props)
```
Comment (tibetsprague) [inline src/routes/LandingPage/LandingPage.connector.js:15]: remove comments

Comment (tibetsprague) [reply src/routes/LandingPage/LandingPage.connector.js:15]: or are you still figuring out how to do this?

File: src/routes/LandingPage/LandingPage.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+  componentDidUpdate () {
+  }
+
+  render () {
+    const { community, location, match, showAbout } = this.props
+
+    if (!community) return <Loading />
+
+    const canView = community.memberCount > 0
```
Comment (tibetsprague) [inline src/routes/LandingPage/LandingPage.js:33]: why is this needed? how could memberCount be 0 and someone be looking at this?

File: src/routes/LandingPage/LandingPage.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    const widgets = community.widgets.filter(w => w.isVisible).sort((a, b) => a.order - b.order)
+    const locationText = community.locationObject.fullText || `${community.locationObject.city}, ${community.locationObject.country}`
+
+    if (!canView) return (<CommunityDetail canClose={false} location={location} match={match} communityId={community.id} />)
```
Comment (tibetsprague) [inline src/routes/LandingPage/LandingPage.js:37]: ?

File: src/routes/LandingPage/LandingPage.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+          <div styleName='title'>
+            <RoundImage url={community.avatarUrl || DEFAULT_AVATAR} large hasBorder={false} />
+            <div>
+              <div styleName='name'>{community.name}</div>
+              <div styleName='location'><Icon name='Location' />{locationText}</div>
+            </div>
+          </div>
+        </div>
+
+        { community.memberCount > 0 && widgets && widgets.map(props => (<Widget {...props} />)) }
```
Comment (tibetsprague) [inline src/routes/LandingPage/LandingPage.js:57]: why the memberCount check again?

File: src/routes/PrimaryLayout/PrimaryLayout.js [hunk 8]
```diff
@@ -166,7 +169,7 @@ export default class PrimaryLayout extends Component {
             <Route path='/:context(all)/:view(topics)' component={AllTopics} />
             <Route path={`/:context(all|public)/:view(map)/${OPTIONAL_POST_MATCH}`} exact component={MapExplorer} />
             <Route path={`/:context(all|public)/:view(map)/${OPTIONAL_COMMUNITY_MATCH}`} exact component={MapExplorer} />
-            <Route path={`/:context(all|public)/${OPTIONAL_POST_MATCH}`} exact component={Feed} />
+            <Route path={`/:context(all|public)/:view(stream|event|project)/${OPTIONAL_POST_MATCH}`} exact component={Feed} />
```
Comment (tibetsprague) [inline src/routes/PrimaryLayout/PrimaryLayout.js:172]: will have to see/understand how this plays with the changes pospi made. and think through our routing in general

File: src/routes/PrimaryLayout/PrimaryLayout.js [hunk 9]
```diff
@@ -206,8 +211,11 @@ export default class PrimaryLayout extends Component {
         </div>
         <div styleName={cx('detail', { hidden: !hasDetail })} id={DETAIL_COLUMN_ID}>
           <Switch>
-            {detailRoutes.map(({ path, component }) =>
-              <Route path={path} component={component} key={path} />)}
+            {detailRoutes.map(({ path, component }) => {
+              return isAboutPath(location.pathname)
```
Comment (tibetsprague) [inline src/routes/PrimaryLayout/PrimaryLayout.js:215]: this feels weird, shouldnt you just list the component as CommunityDetail below in the detailRoutes?

File: src/routes/PrimaryLayout/PrimaryLayout.js [hunk 10]
```diff
@@ -252,6 +263,8 @@ const postEditorRoutes = [
   { path: `/:context(c)/:slug/${REQUIRED_NEW_POST_MATCH}` },
   { path: `/:context(c)/:slug/${REQUIRED_EDIT_POST_MATCH}` },
   { path: `/:context(c)/:slug/m/:personId/${REQUIRED_EDIT_POST_MATCH}` },
+  { path: `/:context(c)/:slug/:view(project|event|stream)/${REQUIRED_NEW_POST_MATCH}` },
```
Comment (tibetsprague) [inline src/routes/PrimaryLayout/PrimaryLayout.js:266]: why is this needed now? does seem like we should talk through routing, the group stuff will also be tricky to merge with this

File: src/routes/PrimaryLayout/components/Navigation/Navigation.js [hunk 11]
```diff
@@ -65,22 +74,28 @@ export default function Navigation (props) {
     }
   ])
 
+  if (!isCommunityPath(location.pathname)) links.shift()
+
   const collapserState = collapsed ? 'collapser-collapsed' : 'collapser'
 
+  const canView = !community || community.memberCount !== 0
```
Comment (tibetsprague) [inline src/routes/PrimaryLayout/components/Navigation/Navigation.js:81]: why would canView be true if !community?

File: src/util/navigation.js [hunk 12]
```diff
@@ -8,7 +8,8 @@ import { host } from 'config'
 
 export const HYLO_ID_MATCH = '\\d+'
 export const POST_ID_MATCH = HYLO_ID_MATCH
-export const DEFAULT_POST_TYPE_CONTEXT = 'p'
+export const SLUG_MATCH = '.+'
```
Comment (tibetsprague) [inline src/util/navigation.js:11]: this doesnt seem right, you should at least be checking for not '/' right?

File: src/util/navigation.js [hunk 13]
```diff
@@ -8,7 +8,8 @@ import { host } from 'config'
...
+export const DEFAULT_POST_TYPE_CONTEXT = 'stream'
```
Comment (tibetsprague) [inline src/util/navigation.js:12]: stream is not a post type context, its a view like map. maybe we just need to call all these things views, and get rid of the idea of POST_TYPE_CONTEXT

File: src/components/PostEditor/PostEditor.connector.js (hunk 2 above)
Comment (wushenshen) [reply src/components/PostEditor/PostEditor.connector.js:65]: @tibetsprague I'm not sure but it didn't seem to break things when I took it out

File: src/routes/CommunitySidebar/CommunitySidebar.js (hunk 3 above)
Comment (wushenshen) [reply src/routes/CommunitySidebar/CommunitySidebar.js:27]: This is meant to hide the sidebar when a person views a community they do not have access to... But I think that might be an edge case possibly coming from my local data - I was once a member of a community and left, and so I can access the `c/slug` url. Whereas if I was never a member of a community, it shows the 404 page.

File: src/routes/LandingPage/LandingPage.connector.js (hunk 4 above)
Comment (wushenshen) [reply src/routes/LandingPage/LandingPage.connector.js:15]: This is still a WIP - will remove comments once it's stable

File: src/routes/LandingPage/LandingPage.js (hunk 5 above)
Comment (wushenshen) [reply src/routes/LandingPage/LandingPage.js:33]: Same comment as [above](https://github.com/Hylozoic/hylo-evo/pull/751#discussion_r562070442)

File: src/routes/PrimaryLayout/PrimaryLayout.js (hunk 9 above)
Comment (wushenshen) [reply src/routes/PrimaryLayout/PrimaryLayout.js:215]: Need to pass `communityId` to the component `<CommunityDetail>` which didn't seem possible to do by using the structure in `detailRoutes` below. That's because the CommunityDetail component relies on the id rather than the slug and the id is not available in the routeParams that are passed through.

File: src/routes/PrimaryLayout/components/Navigation/Navigation.js (hunk 11 above)
Comment (wushenshen) [reply src/routes/PrimaryLayout/components/Navigation/Navigation.js:81]: The canView refers to the Navigation component and it visible in network, all, and public contexts where there is no community.

**Comment by lorenjohnson** [comment]
I've rebased this in the PR to the `holonic-architecture` branch. There are still some conflicts to deal with. What else remains for this branch to move forward? @wushenshen @tibetsprague
